src/Framework/Framework/Resources/Scripts/utils/evaluator.ts

```diff
@@ -1,18 +1,51 @@
 import { isObservableArray } from "./knockout";
 import { logError } from "./logging";
 
-export function evaluateOnViewModel(context: any, expression: string): any {
-    // todo: reimplement
-    let result;
-    if (context && context.$data) {
-        result = new Function("$context", "with($context) { with ($data) { return " + expression + "; } }")(context);
-    } else {
-        result = new Function("$context", "var $data=$context; with($context) { return " + expression + "; }")(context);
+const shouldBeConvertedFromDataContext = (currentLevel: any, remainingParts: string[]): boolean => {
+    if (currentLevel["$data"] == undefined) {
+        return false;
     }
-    if (result && result.$data) {
-        result = result.$data;
+
+    return remainingParts.filter(i => i.startsWith("$")).length === 0;
+};
+
+
+
+/**
+ * Traverses provided context according to given path.
+ * @example / - returns context
+ * @example "" or null - returns context
+ * @example /exampleProp/A - returns prop A located withing property exampleProp located at provided context
+ * @example /exampleProp/B/1 - returns second item from collection B located within property exampleProp located at provided context
+ * @returns found property as unwrapped object
+ */
+export function traverseContext(context: any, path: string): any 
+{
+    if (path == null)
+        path = "";
+
+    var parts = path.split(/[/[\.\]]+/);
+    var currentLevel = context;
+    var currentPath = "";
+    for (var i = 0; i < parts.length; i++) {
+        if (shouldBeConvertedFromDataContext(currentLevel, parts.slice(i))) {
```

Comment:
There are some use cases when we get $data and $rawData in path.
- `$data`
   - Is received when we are validating current context (`validate-this`). This could be replaced by root path ('/').
- `$rawData`
   -   If I'm correct this comes as part of validation path with changed data context (@acizmarik knows more about the server part). We were not able to get of this unfortunately.

Reply:
@MichalTichy so, can we remove this?

Reply:
Removed